tests/unitTests/test.h

```diff
@@ -774,6 +882,47 @@ class TestMaterialModel {
             std::cout << std::endl;
         }
     }
+
+    // Function to test rho, beta, drho/dp, and dbeta/dp from g_vol_pen() against 
+    // reference solution. These are required for treating a volumetric penalty
+    // term in the ustruct formulation
+    // ARGS:
+    // - p: Pressure
+    // - rho0: Initial solid density
+    // - rho_ref: Reference solution for rho
+    // - beta_ref: Reference solution for beta
+    // - drhodp_ref: Reference solution for drho/dp
+    // - dbetadp_ref: Reference solution for dbeta/dp
+    // - rel_tol: Relative tolerance for comparing rho and beta with reference solutions
+    // - abs_tol: Absolute tolerance for comparing rho and beta with reference solutions
+    // - verbose: Show values of p, rho, beta, rho_ref, beta_ref
+    void testRhoBetaAgainstReference(double p, double rho0, double rho_ref, double beta_ref, double drhodp_ref, double dbetadp_ref, double rel_tol, double abs_tol, bool verbose = false) {
+        double rho, beta, drhodp, dbetadp;
+        double Ja = 1.0; // Active strain Jacobian (not used in this function)
+
+        // Compute rho, beta, drhodp, dbetadp from g_vol_pen()
+        g_vol_pen(p, rho0, rho, beta, drhodp, dbetadp, Ja);
+
+        // Compare rho, beta, drho, dbeta with reference solutions
+        EXPECT_NEAR(rho, rho_ref, fmax(abs_tol, rel_tol * fabs(rho_ref)));
+        EXPECT_NEAR(beta, beta_ref, fmax(abs_tol, rel_tol * fabs(beta_ref)));
+        EXPECT_NEAR(drhodp, drhodp_ref, fmax(abs_tol, rel_tol * fabs(drhodp_ref)));
+        EXPECT_NEAR(dbetadp, dbetadp_ref, fmax(abs_tol, rel_tol * fabs(dbetadp_ref)));
+
```

Comment:
Optional: Could we write our own "near" function a la [numpy isclose](https://numpy.org/doc/stable/reference/generated/numpy.isclose.html)?
```
absolute(a - b) <= (atol + rtol * absolute(b))
```

Reply:
Makes sense!